File: SpaClient/src/app/models/issue.ts
import { Label } from './label';
import { PhaseListItem } from './phase-list-item';
import { IssueUser } from './issue-user';
import { User } from './user';
import { Comment } from './comment';
import { Attachment } from './attachment';

export interface Issue {
  id: string;
  name: string;
  description: string;
  dueDate: string;
  createdAt: string;
  issueType: string;
  priority: number;
  status: string;

  createdBy: User;
  phase: PhaseListItem;

  labels: Label[];
  issuedTo: IssueUser[];
  comments: Comment[];
  attachments: Attachment[];
}

File: Api/Models/Issue.cs
using Api.Helpers.DTOs;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Api.Models
{
    public class Issue
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public string Id { get; set; }

        [Required]
        [MaxLength(255)]
        public string Name { get; set; }

        [Required]
        public string Description { get; set; }
        
        public string DueDate { get; set; }
        
        [Required]
        public string CreatedAt { get; set; }
        
        [Required]
        public string IssueType { get; set; }

        [Required]
        public int Priority { get; set; }

        [Required]
        public string Status { get; set; }

        [Required]
        public string CreatedById { get; set; }
        public User CreatedBy { get; set; }

        [Required]
        public string PhaseId { get; set; }

        [Required]
        public Phase Phase { get; set; }

        public ICollection<IssueLabel> IssueLabels { get; set; }
        public ICollection<IssueUser> IssuedTo { get; set; }
        public ICollection<Comment> Comments { get; set; }

        public Issue()
        {
            IssueLabels = new List<IssueLabel>();
            IssuedTo = new List<IssueUser>();
        }
    }
}
